dotnet/src/SemanticKernel.Core/Data/VolatileVectorStoreRecordCollection.cs

```diff
@@ -14,35 +13,31 @@ namespace Microsoft.SemanticKernel.Data;
 /// <summary>
 /// Service for storing and retrieving vector records, that uses an in memory dictionary as the underlying storage.
 /// </summary>
+/// <typeparam name="TKey">The data type of the record key.</typeparam>
 /// <typeparam name="TRecord">The data model to use for adding, updating and retrieving data from storage.</typeparam>
 [Experimental("SKEXP0001")]
 #pragma warning disable CA1711 // Identifiers should not have incorrect suffix
-public sealed class VolatileVectorStoreRecordCollection<TRecord> : IVectorStoreRecordCollection<string, TRecord>
+public sealed class VolatileVectorStoreRecordCollection<TKey, TRecord> : IVectorStoreRecordCollection<TKey, TRecord>
```

Comment:
Should this be:

`public sealed class VolatileVectorStoreRecordCollection<TRecord> : IVectorStoreRecordCollection<object, TRecord>`

Reply:
This would cause the users to only be able to use objects and would make TKey on the class redundant.  We do want TKey to be on the interface, so users can have strongly typed keys.